src/apis/HTTPClient.ts

```diff
@@ -0,0 +1,80 @@
+import axios, {
+  AxiosError,
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosResponse
+} from "axios";
+
+export default class HTTPClient {
+  private client: AxiosInstance;
+  private customOnError: ((error: any) => void) | undefined;
+
+  constructor(baseUrl: string, secure: boolean) {
+    this.client = axios.create({
+      baseURL: baseUrl,
+      headers: {
+        "Cache-Control": "no-store, no-cache, must-revalidate",
+        Pragma: "no-cache",
+        Expires: 0
+      }
+    });
+
+    if (secure) {
+      this.client.interceptors.request.use(
+        async (config) => {
+          const token = window.localStorage.getItem("token");
+          if (token) config.headers.Authorization = `Bearer ${token}`;
+          return config;
+        },
+        (error) => {
+          return Promise.reject(error);
+        }
+      );
+    }
+  }
+
+  protected get = <T>(
+    url: string,
+    axiosReqConfig?: AxiosRequestConfig
+  ): Promise<T> =>
+    this.client
+      .get(url, axiosReqConfig)
+      .then(this.onSuccess)
+      .catch(this.onError) as unknown as Promise<T>;
```

Comment:
Instead of typecasting `as unknown as Promise<T>`, use `this.client.get<T>(url, ...etc`

Reply:
Already done